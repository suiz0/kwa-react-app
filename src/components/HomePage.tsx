import React from 'react';

const HomePage = (props) => {
    const content = (
      <div className="bx--grid">
        <div className="bx--row">
            <div className="bx--col">{props.t ? props.t('Home Page'): 'Home Page'}</div>
        </div>
      </div>
    )
  
    return content;
}

export default HomePage;