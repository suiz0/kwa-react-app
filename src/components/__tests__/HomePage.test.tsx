import React from 'react';
import HomePage from '../HomePage';
import { shallow , configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

let component;

test('renders without crashing', () => {
    component = shallow(<HomePage authUser={{authenticated:false}} />);
    expect(component).toMatchSnapshot();
});

describe('Composition tests', () => {

    beforeEach(()=> {
        component = shallow(<HomePage authUser={{authenticated:false}} />);
    });

    test('renders a div with bx--grid class', ()=> {
        expect(component.find("div.bx--grid")).toHaveLength(1);
    });

    test('renders page content', ()=> {
        const PageContent = <div className="bx--col">Home Page</div>
        expect(component.contains(PageContent)).toEqual(true);
    });
});
