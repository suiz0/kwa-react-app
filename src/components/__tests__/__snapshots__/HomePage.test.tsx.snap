// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="bx--grid"
>
  <div
    className="bx--row"
  >
    <div
      className="bx--col"
    >
      Home Page
    </div>
  </div>
</div>
`;
